oshi-core/src/test/java/oshi/hardware/CentralProcessorTest.java

```diff
@@ -49,45 +49,60 @@ public void testCentralProcessor() {
+        assertNotNull("Processor Identifier's vendor shouldn't be Null", pi.getVendor());
+        assertTrue("Processor Identifier's vendor frequency should either be -1, or a greater then 0",
+                pi.getVendorFreq() == -1 || pi.getVendorFreq() > 0);
+        assertNotNull("Processor Identifier's name shouldn't be Null", pi.getName());
+        assertNotNull("Processor Identifier's identifier shouldn't be Null", pi.getIdentifier());
+        assertNotNull("Processor Identifier's ID shouldn't be Null", pi.getProcessorID());
+        assertNotNull("Processor Identifier's stepping shouldn't be Null", pi.getStepping());
+        assertNotNull("Processor Identifier's model shouldn't be null", pi.getModel());
+        assertNotNull("Processor Identifier's family shouldn't be null", pi.getFamily());
+        assertFalse("Processor Identifier's micro-architecture shouldn't be blank",
+                Util.isBlank(pi.getMicroarchitecture()));
 
         long[] ticks = p.getSystemCpuLoadTicks();
         long[][] procTicks = p.getProcessorCpuLoadTicks();
-        assertEquals(ticks.length, TickType.values().length);
+        assertEquals("System should have the same amount of cpu-load-tick counters as there are TickType values",
+                ticks.length, TickType.values().length);
 
         Util.sleep(500);
 
-        assertTrue(p.getSystemCpuLoadBetweenTicks(ticks) >= 0 && p.getSystemCpuLoadBetweenTicks(ticks) <= 1);
-        assertEquals(3, p.getSystemLoadAverage(3).length);
+        assertTrue("System's cpu load between ticks should be inclusively between 0 and 1",
+                p.getSystemCpuLoadBetweenTicks(ticks) >= 0 && p.getSystemCpuLoadBetweenTicks(ticks) <= 1);
+        assertEquals("System's load averages length for 3 elements should equal 3", 3,
+                p.getSystemLoadAverage(3).length);
 
-        assertEquals(p.getProcessorCpuLoadBetweenTicks(procTicks).length, p.getLogicalProcessorCount());
+        assertEquals("Central Processor's cpu load between ticks should equal the logical processor count",
+                p.getProcessorCpuLoadBetweenTicks(procTicks).length, p.getLogicalProcessorCount());
         for (int cpu = 0; cpu < p.getLogicalProcessorCount(); cpu++) {
-            assertTrue(p.getProcessorCpuLoadBetweenTicks(procTicks)[cpu] >= 0
-                    && p.getProcessorCpuLoadBetweenTicks(procTicks)[cpu] <= 1);
-            assertEquals(p.getProcessorCpuLoadTicks()[cpu].length, TickType.values().length);
+            assertTrue("Central Processor's cpu number " + cpu + "'s load between ticks should be inclusively between 0 and 1",
+                    p.getProcessorCpuLoadBetweenTicks(procTicks)[cpu] >= 0
+                            && p.getProcessorCpuLoadBetweenTicks(procTicks)[cpu] <= 1);
+            assertEquals("Central Processor's cpu number " + cpu + " should have the same amount of cpu-load-tick" +
+                            "counters as there are TickType values",
```

Comment:
not sure I like manually splitting the string here.  Does this miss a space this way?

Reply:
It does miss a space at the end, I tried making it not as long to make it more readable. I don't like manually splitting lines either but to be honest I haven't done java in a while and forgot a better way. Any suggestion?

Reply:
I'd just leave it as-is (a longer string). Doesn't hurt to scroll off the screen.